.github/workflows/pull_request.yml

```diff
@@ -31,44 +33,22 @@ jobs:
           key: ${{ runner.os }}-yarn-${{ hashFiles('yarn.lock') }}
       - name: Install yarn deps
         run: yarn install
-      - name: Install Pipenv
-        run: pip install pipenv==2018.11.26
-      - name: Cache virtualenv
-        id: cache-virtualenv
-        uses: actions/cache@v1
-        with:
-          path: ~/.local/share/virtualenvs/
-          key: ${{ runner.os }}-${{ env.PYTHON_VERSION }}-virtualenvs-${{ hashFiles('Pipfile.lock') }}
```

Comment:
Since i combined Lint job in same job so caching is not required. I will wait for other devs if they want Lint and test as separate jobs then caching is required.

Reply:
My preference is to have these separate, so it is visually better in the PR, as it is easier to see what failed without going into the logs. Also, we hopefully will eventually use poetry in runner which would benefit from cache, so I think it would be easier to be consistent between our repos. As discussed yesterday, I'll have a play with it soon to see if I can get cache to play nice or whether I get similar issues you faced.

Reply:
I have a working version with cache here. https://github.com/ONSdigital/eq-questionnaire-validator/actions/runs/1783626063/workflow